src/core/qgsmaplayer.cpp

```diff
@@ -3253,7 +3253,24 @@ QString QgsMapLayer::generalHtmlMetadata() const
         continue;
 
       const QVariant propValue = customProperty( key );
-      metadata += QStringLiteral( "<tr><td class=\"highlight\">%1</td><td>%2</td></tr>" ).arg( key.toHtmlEscaped(), propValue.toString().toHtmlEscaped() );
+      QString stringValue = QString();
+      if ( propValue.canConvert<QStringList>() && propValue.toStringList().count() > 1 )
+      {
+        for ( const QString &s : propValue.toStringList() )
+        {
+          stringValue += s.toHtmlEscaped() + QStringLiteral( "<br>" );
+        }
+      }
+      else if ( propValue.canConvert<QString>() || propValue.isNull() )
```

Comment:
Similarly, don't use canConvert here.

Also, we shouldn't use QVariant::isNull to test for null, rather QgsVariantUtils::isNull, or you'll get unpredictable results on Qt 6 builds.